harness/determined/experimental/core_v2/_core_v2.py

```diff
@@ -121,6 +137,7 @@ def _set_globals() -> None:
 
 def _init_context(
```

Comment:
this is an internal/private method, so i think it makes sense to no longer pass in `defaults` and `unmanaged` here. 

since we've deprecated defaults and unmanaged, can we just do the merge logic (construct the new `config` object) upfront (in `init()` and `init_context()`)? this would reflect our intention to remove the public facing parameters, and would make the actual deprecation much easier. currently, we're saying it's deprecated but we're actually still using the objects, and whoever deprecates them later on would need to know context of detached mode to properly deprecate them.